### ISSUE TITLE
Make axis names accessible from layers

### ISSUE THREAD

**Issue by GenevieveBuckley** [napari/napari#1144]
## 🚀 Feature
Make axis names accessible from layers

## Motivation

It's been mentioned twice now that it would be desirable to be able to access the axis names from individual layers. Right now, axis names live only on the viewer. 

This would be particularly useful when writing csv files from layer data, since then we could have more appropriately named column headings. Right now, column headings are along the lines of `axis-0`, `axis-1`, ... instead of something more meaningful.

## Additional context

https://github.com/napari/napari/pull/941#issuecomment-581511735
>>    Is there a better row header for the points csv than "id, dimension_0, dimension_1, ..."? Is it possible/should we try to tie the csv headings to the actual dimension names (xyztc)?
>
> I like the idea of taking the actual dimension names which I think can be found in viewer.dims.axis_labels, but these don't live in the actual layer (i.e. layer.dims.axis_labels will just have default values), so we might need to give this more thought.

https://github.com/napari/napari/pull/1104#discussion_r407557349
>>    the coordinates data should include axis names as the column headers, now that we have them.
>
> we don't have them - right now they reside on the viewer and not on the layer and so don't get passed in through the meta dict. This is true for both reading and writing.
>
> @GenevieveBuckley took the approach of dim-0 through dim-N in #941 which is where i took this from. I can copy over that approach here, it will make the transition when we have axis labels easier.

**Comment by TimMonko** [comment]
@ianhi and this one is also relevant to napari/island-dispatch#16

**Comment by psobolewskiPhD** [comment]
Just a reminder that now Layers do have `Layer.axis_labels`: https://github.com/napari/napari/blob/fe71d498e84c22fd0d159bcff01aaecb5226e887/src/napari/layers/base/base.py#L817
but they are distinct/unconnected to `Dims.axis_labels` https://github.com/napari/napari/blob/fe71d498e84c22fd0d159bcff01aaecb5226e887/src/napari/components/dims.py#L95

**Comment by TimMonko** [comment]
Yes Yes! We need a meta issue for this :) I'm realizing (and @ianhi DMed me the same) that we kind of have many open issues on this general topic.

**Comment by psobolewskiPhD** [comment]
Need a pinned meta issue of meta issues.

**Comment by jni** [comment]
I think this specific issue was closed by #6979
